yt/java/ytsaurus-client/src/main/java/tech/ytsaurus/client/StartOperationHelpers.java

```diff
@@ -0,0 +1,47 @@
+package tech.ytsaurus.client;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import tech.ytsaurus.ysontree.YTree;
+import tech.ytsaurus.ysontree.YTreeNode;
+
+import java.net.URI;
+import java.util.Arrays;
+
+public class StartOperationHelpers {
+    private static final Logger logger = LoggerFactory.getLogger(StartOperationHelpers.class);
+
+    // Copied from `BaseLayerDetector.guess_base_layers`
+    public static void guessBaseLayers(YTreeNode spec) {
```

Comment:
I propose to move this logic to [MapperOrReducerSpec](https://github.com/ytsaurus/ytsaurus/blob/main/yt/java/ytsaurus-client/src/main/java/tech/ytsaurus/client/operations/MapperOrReducerSpec.java#L44).